File: .make-it/prototype/stubs/androidx/room/RoomDatabase.java
package androidx.room;

import androidx.sqlite.SQLiteDriver;

/** Minimal stub of androidx.room.RoomDatabase with the nested Builder.setDriver(SQLiteDriver). */
public class RoomDatabase {
  public static class Builder {
    public Builder setDriver(SQLiteDriver driver) {
      System.out.println(
          "    setDriver(...) received runtime type = "
              + (driver == null ? "null" : driver.getClass().getName()));
      return this;
    }
  }
}

File: plugin-build/src/test/kotlin/androidx/room/RoomDatabase.kt
package androidx.room

import androidx.sqlite.SQLiteDriver

abstract class RoomDatabase {
  interface SetDriverReceiver {
    fun setDriver(driver: SQLiteDriver): Builder<*>
  }

  open class Builder<T : RoomDatabase> : SetDriverReceiver {
    override fun setDriver(driver: SQLiteDriver): Builder<T> = this
  }
}
